### PR TITLE
glibc: clean up install, add C.UTF-8 by default

### PR DESCRIPTION
**Pull request by ahesford**
We should really provide the `C.UTF-8` locale by default. I believe that the best way to do so is by removing it entirely from `/etc/default/libc-locales` processing and install it, as part of the main `glibc` package, in `/usr/lib/locales` like Arch does.

This currently trips the package linter, so either
- We can make an exception for glibc in the linter, or
- Rather than include the locale files in the package, we can add an `INSTALL` script to the base package to write the `C.utf-8` locale outside of an archive, where it will always be reachable.

I favor the first because it is less wasteful; why should everybody have to generate a locale that we expect everybody to have available by default?

A third option is to modify the `glibc-locales` `INSTALL` script to either add a separate, non-archived `C.utf-8` locale directory, or just ensure that `C.UTF-8` is always included in the locale processing. Both seem even more kludgey than doing the right thing by default.

Thoughts? @void-linux/pkg-committers 

While I was at it, I cleaned up the install function a bit.

### CODE REVIEW

**Comment by ahesford** [comment]
For bikeshedding, I added the proposed linter exception.

File: srcpkgs/glibc/template [hunk 1]
```diff
@@ -123,23 +123,28 @@ do_build() {
...
 	mkdir -p ${DESTDIR}/etc
 	echo "include /etc/ld.so.conf.d/*.conf" > ${DESTDIR}/etc/ld.so.conf
 
-	cd build
-	make install_root=${DESTDIR} install
+	( cd build && make install_root=${DESTDIR} install )
+
+	# The C.UTF-8 locale should always be available
+	vmkdir usr/lib/locale
+	I18NPATH=./localedata localedef --prefix="${DESTDIR}" \
```
Comment (oreo639) [inline srcpkgs/glibc/template:137]: Use the system localedef when cross compiling and the recently built localedef when native compiling.
Also, make sure to pass `--big-endian` or `--little-endian` when cross compiling.

See: https://salsa.debian.org/glibc-team/glibc/-/blob/6c8b05d8a7138593c80908d5fe5b928d2ac038d6/debian/rules.d/build.mk#L340-358

File: srcpkgs/glibc/template [hunk 2]
```diff
@@ -123,23 +123,28 @@ do_build() {
...
 	vinstall ${FILESDIR}/libc-locales 644 etc/default
-	cp ${wrksrc}/localedata/SUPPORTED ${wrksrc}/build
-	sed -i	-e "s|/| |g" \
+	sed -e "s|/| |g" \
 		-e 's|\\| |g' \
 		-e "s|SUPPORTED-LOCALES=||" \
-		-e "s|^|#|g" ${wrksrc}/build/SUPPORTED
-	cat ${wrksrc}/build/SUPPORTED >> ${DESTDIR}/etc/default/libc-locales
+		-e "s|^|#|g" \
+		-e "/#C\.UTF-8 /d" localedata/SUPPORTED \
```
Comment (oreo639) [inline srcpkgs/glibc/template:146]: imo, probably fine to leave it?

**Comment by ahesford** [comment]
Changes adopted. For native builds, we can just use the make target; for cross builds, we will fall back to the host utility with proper endianness.

**Comment by oreo639** [comment]
Using the system `localedef` command is a bit problematic when cross-compiling from musl.
Maybe it would be good to use the qemu build helper instead?
(for that either we can define `LOCALEDEF` manually like Debian, just call localedef manually with `vtargetrun`, or use `vsed -i localedata/Makefile -e "s|\$(rtld-prefix)|qemu-${XBPS_TARGET_QEMU_MACHINE}-static \$(rtld-prefix)|"`)

**Comment by ahesford** [comment]
We don't generally support cross-compiling across C libraries, and I'm not sure about the wisdom of requiring `qemu-user-static` in a bootstrap package like glibc.

**Comment by oreo639** [comment]
Yeah.

**Pull request merged**